src/Roots.jl

```diff
@@ -49,72 +56,41 @@ struct RootsParameters{FT <: AbstractFloat}
     K_max_stem_moles::FT
 end
 
-
 """
-    RootsModel
+    RootsModel{FT, PS, D, RE, T, B} <: AbstractModel{FT}
 
 Defines, and constructs instances of, the RootsModel type, which is used
 for simulation flow of water to/from soil, along roots of different depths,
 along a stem, to a leaf, and ultimately being lost from the system by
 transpiration. 
 
 This model can be used in standalone mode by prescribing the transpiration rate
-and soil pressure at the root tips (configuration of type `RootsConfiguration`),
-or with a dynamic soil model (boundary exchanges of type TBD).
+and soil pressure at the root tips, or with a dynamic soil model.
 
 The RootModel domain must be of type `AbstractPlantDomain`.
 """
-struct RootsModel{FT, PS, D, B} <: AbstractVegetationModel{FT}
+struct RootsModel{FT, PS, D, RE, T} <: AbstractVegetationModel{FT}
     param_set::PS
     domain::D
-    configuration::B
+    root_extraction::RE
+    transpiration::T
```

Comment:
In the soil model, we have `sources` and `boundary_conditions`, while here we have fields for what seem to be specific processes. Does it make sense to group these processes in a `sources` tuple that is stored in the model?

Reply:
that's a good question! I dont know the answer....

followup questions:
- is a single term on the RHS of an ODE called a source?
- is this the terminology they use in that field, regardless of what mathematicians think?
- the two "sources" appear in different ODEs - would we then need to specify which variable the source terms apply to (in the tuple)? we may need to do this with soil anyways when we add in heat, so not an issue really...
- is it clearer to group by physical concept, rather than function (they are both added in to RHS's)?

Reply:
made https://github.com/CliMA/ClimaLSM.jl/issues/19